src/templates/home.jsx

```diff
@@ -20,27 +21,44 @@ const Home = ({
     wpPage: { seo, acf: data },
     allWpPost,
   },
-  pageContext: { locale, pageUrls, menus, globalFields },
-}) => (
-  <MainLayout
-    locale={locale}
-    seo={seo}
-    pageUrls={pageUrls}
-    menus={menus}
-    globalFields={globalFields}
-  >
-    <Hero {...data.hero} />
-    <Advantages {...data.advantages} />
-    <SolutionsProducts {...data.solutionsProducts} />
-    <News {...data.news} {...allWpPost} readMoreText={translations[locale].blog.postCtaButton} />
-    <Technologies {...data.technologies} />
-    <Partners {...data.partners} />
-    <Jobs {...data.jobs} />
-    <Report {...data.report} />
-    <Awards {...data.awards} />
-    <Contact locale={locale} />
-  </MainLayout>
-);
+  pageContext: { locale, pageUrls, menus, globalFields, upcomingEvents },
+}) => {
+  const [featuredUpcomingEvents, setFeaturedUpcomingEvents] = useState([]);
+  useEffect(() => {
+    if (!upcomingEvents) return;
+    const events = upcomingEvents.reverse().slice(0, 3);
+    setFeaturedUpcomingEvents(events);
+  }, [upcomingEvents]);
```

Comment:
There is no need actually in `useState` - we use it as a holder variable - better to leverage `useMemo` instead of `useState` + `useEffect`